src/error.rs

```diff
@@ -0,0 +1,125 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::fmt;
+use std::result;
+
+use failure::{Backtrace, Context, Fail};
+use rocket::http::{Status};
+use rocket::response::{Responder, Response};
+use rocket::{self, response, Request};
+use rocket_contrib::Json;
+
+use logging::MozlogLogger;
+
+pub type HandlerResult<T> = result::Result<T, HandlerError>;
+
+#[derive(Debug)]
+pub struct HandlerError {
+    inner: Context<HandlerErrorKind>,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Fail)]
+pub enum HandlerErrorKind {
```

Comment:
It's no biggie but fwiw I'm not a huge fan of the `...Kind` suffix in this name. Is it a Rust convention or just something we opted for ourselves?

Reply:
Hm, I followed this tutorial from the `failure` book -> https://boats.gitlab.io/failure/error-errorkind.html

It's the naming convention that they use there, but I guess we could name it something we think is more semantic. What did you have in mind?

Reply:
No worries, if it's an idiomatic Rust thing I'm happy!